core/trino-main/src/main/java/io/trino/sql/analyzer/TypeSignatureTranslator.java

```diff
@@ -300,6 +301,19 @@ static DataType toDataType(TypeSignature typeSignature)
         }
     }
 
+    private static boolean requiresDelimiting(String identifier)
```

Comment:
This is ok for now, but technically, every name should be delimited. Once https://github.com/trinodb/trino/issues/17 is fixed, any names coming from a connector (including column types and field names in row types) will be considered as "already canonicalized", so they'll need to be surrounded in quotes when rendering as SQL to ensure proper matching semantics.